recipes/arrow/all/conanfile.py

```diff
@@ -261,6 +264,11 @@ def validate(self):
             if self.dependencies["jemalloc"].options.enable_cxx:
                 raise ConanInvalidConfiguration("jemmalloc.enable_cxx of a static jemalloc must be disabled")
 
+        if self.options.with_thrift and not self.options.with_zlib:
```

Comment:
There are more checks in `ThirdpartyToolchain.cmake` that might be worth to reflect in the recipe, instead of relying on autodiscovery

Reply:
That file is over 5000 lines long :D. I do not have the bandwidth to read through it to see what's going on.
If you have some specific checks in mind?

Otherwise I suggest we leave this to be addressed in another PR.

Reply:
This is the only one that stood out to me, so I'm happy with keeping that on a follow-up PR, yes :)